src/board_controller/openbci/openbci_serial_board.cpp

```diff
@@ -144,36 +143,68 @@ int OpenBCISerialBoard::prepare_session ()
     int port_open = open_port ();
     if (port_open != STATUS_OK)
     {
+        delete serial;
+        serial = NULL;
         return port_open;
     }
 
     int set_settings = set_port_settings ();
     if (set_settings != STATUS_OK)
     {
+        delete serial;
```

Comment:
BTW we dont need it here, if session is failed board_controller will call a destructor which will delete serial object(but it may be unclear for somebody)